src/ophys_etl/modules/segmentation/merge/candidates.py

```diff
@@ -0,0 +1,149 @@
+        pixel_distance: float,
+        rois_to_ignore: Optional[Set[int]],
+        output_list: multiprocessing.managers.ListProxy) -> None:
+    """
+    Find all of the abutting ROIs in a list of OphysROIs
+
+    Parameters
+    ----------
+    roi_id_pair_list: List[Tuple[int, int]]
+        Pair of roi_ids to consider for merging
+
+    roi_lookup: Dict[int, OphysROI]
+        Maps roi_id to OphysROI
+
+    pixel_distance: float
+       The maximum distance from each other two ROIs can be at
+       their nearest point and still be considered to abut
+
+    rois_to_ignore: Optional[Set[int]]
+       Optional set of ints specifying roi_id of ROIs not to consider
+       when looking for pairs. Note: a pair will only be ignored
+       if *both* ROIs are in rois_to_ignore. If one of them is not,
+       the pair is valid (default: None)
+
+    output_list: multiprocessing.managers.ListProxy
+        List where valid merger candidates are stored
+
+    Returns
+    -------
+    None
+        valid merger candidates are stored in output_list
+    """
+    for roi_id_pair in roi_id_pair_list:
+        if roi_id_pair[0] == roi_id_pair[1]:
+            continue
+
+        if rois_to_ignore is not None:
+            if roi_id_pair[0] in rois_to_ignore:
+                if roi_id_pair[1] in rois_to_ignore:
+                    continue
+
+        roi0 = roi_lookup[roi_id_pair[0]]
+        roi1 = roi_lookup[roi_id_pair[1]]
+        if do_rois_abut(roi0, roi1, pixel_distance=pixel_distance):
+            output_list.append(roi_id_pair)
+    return None
+
+
+def find_merger_candidates(
+        roi_list: List[OphysROI],
+        pixel_distance: float,
+        rois_to_ignore: Optional[Set[int]] = None,
+        n_processors: int = 8) -> List[Tuple[int, int]]:
+    """
+    Find all the pairs of abutting ROIs in a list of OphysROIs.
+    Return a list of tuples like (roi_id_0, roi_id_1) specifying
+    the ROIs that abut.
+
+    Parameters
+    ----------
+    roi_list: List[OphysROI]
+
+    pixel_distance: float
+       The maximum distance from each other two ROIs can be at
+       their nearest point and still be considered to abut
+
+    rois_to_ignore: Optional[Set[int]]
+       Optional set of ints specifying roi_id of ROIs not to consider
+       when looking for pairs. Note: a pair will only be ignored
+       if *both* ROIs are in rois_to_ignore. If one of them is not,
+       the pair is valid (default: None)
+
+    n_processors: int
+       Number of cores to use (this method uses multiprocessing since, for
+       full fields of view, there can be tens of millions of pairs of
+       ROIs to consider)
+
+   Returns
+   -------
+   output: List[Tuple[int, int]]
+       List of tuples of roi_ids specifying pairs of abutting ROIs
+    """
+    lookup = {}
+    roi_id_list = []
+    for roi in roi_list:
+        lookup[roi.roi_id] = roi
+        roi_id_list.append(roi.roi_id)
+
+    mgr = multiprocessing.Manager()
```

Comment:
I think before this heads to `main` we need to replace the `Manager/Process/_winnow` system with `Pool`
no change requested right now.

Reply:
The passing of return values into the output_dict argument lends itself to less explicit typing.

Reply:
I'll open a ticket to get this working with `multiprocessing.Pool`

Reply:
#278

Reply:
thank you